### PR TITLE
Biruk/57 answer repository

### LINKED ISSUE #57: Answers Repository

### PR DESCRIPTION
**Pull request by Biruk-Tassew**

### CODE REVIEW

File: qelem/lib/infrastructure/answer/answer_repoistory.dart
Comment (hello-world-was-taken) [inline qelem/lib/infrastructure/answer/answer_repoistory.dart]:
```suggestion
  Future<Either<Answer>> updateAnswer(
      AnswerFormDto answerFormDto, int answerId, String content) async {
    try {
      AnswerDto updatedAnswer =
          await answerApi.updateAnswer(answerFormDto, answerId, content);
      return Either(val: updatedAnswer.toAnswer());
```

**Review by hello-world-was-taken — CHANGES_REQUESTED** [review]

File: qelem/lib/infrastructure/answer/answer_repoistory.dart
Comment (hello-world-was-taken) [inline qelem/lib/infrastructure/answer/answer_repoistory.dart]: How would the UI update when there is an upvote, if there is not response (void response) from the backend?

File: qelem/lib/presentation/pages/question/widgets/question_card.dart
Comment (hello-world-was-taken) [inline qelem/lib/presentation/pages/question/widgets/question_card.dart]: This will work once it is merged with my branch, so, no worries. Or you could use "assets/images/default_profile_picture.jpeg" or "assets/abebe.jpeg" for the image and it will work. I am assuming that is the problem though.

File: qelem/lib/infrastructure/answer/answer_repoistory.dart
Comment (Biruk-Tassew) [reply qelem/lib/infrastructure/answer/answer_repoistory.dart]: what if i send .getAnswerById to the api and return the response?

Comment (hello-world-was-taken) [reply qelem/lib/infrastructure/answer/answer_repoistory.dart]: @Biruk-Tassew see my pull request [question-repository](https://github.com/s9uidward/Qelem/pull/67/files#diff-2046dd6c6e9f333e77f26daaecb03c880a75b815a86bd59f4b5b2b9d361fe72a). There is a huge similarity between this and the one I did. If you have other suggestion, do let me know as it would also help me.

**Review by nawrazi — APPROVED** [review]

File: qelem/lib/domain/answer/answer.dart
Comment (brukted) [inline qelem/lib/domain/answer/answer.dart]: Use the domain model

File: qelem/lib/infrastructure/answer/answer_api.dart
Comment (brukted) [inline qelem/lib/infrastructure/answer/answer_api.dart]:
```suggestion
        await _httpClient.post('answers', body: json.encode(answerFormDto.toJson()));
```

File: qelem/lib/infrastructure/answer/answer_repoistory.dart
Comment (brukted) [inline qelem/lib/infrastructure/answer/answer_repoistory.dart]: handle network exception

File: qelem/lib/infrastructure/answer/answer_repoistory.dart
Comment (brukted) [inline qelem/lib/infrastructure/answer/answer_repoistory.dart]: create domain answer form class.

File: qelem/lib/infrastructure/answer/answer_repoistory.dart
Comment (brukted) [inline qelem/lib/infrastructure/answer/answer_repoistory.dart]: similar thing with the form dto

**Review by brukted — CHANGES_REQUESTED** [review]

File: qelem/lib/domain/answer/answer.dart
Comment (Biruk-Tassew) [reply qelem/lib/domain/answer/answer.dart]: for the user? if so ok...

File: qelem/lib/infrastructure/answer/answer_api.dart
Comment (Biruk-Tassew) [reply qelem/lib/infrastructure/answer/answer_api.dart]: encode or decode?

**Review by brukted — APPROVED** [review]

**Comment by brukted** [comment]
@hello-world-was-taken, we are waiting for your approval.

**Comment by hello-world-was-taken** [comment]
> @hello-world-was-taken, we are waiting for your approval.

@brukted, sorry, I didn't get the time to go over it.

**Pull request merged**
